### PR TITLE
feat: add NRF registrations, subcriptions and discovery metrics

### PR DESCRIPTION
**Pull request by dariofaccin**
This PR adds three new 5G specific metrics, `nrf_registrations`, `nrf_subscriptions` and `nrf_nf_instances` to exported metrics via Prometheus.
The scope of the metrics is the following:
* `nrf_registrations`: this metric counts the total number of registrations to NRF; the metric stores the query type (`register`, `update`, `deregister`), the requester NF type and the result (`SUCCESS`, `FAILURE`)
* `nrf_subscriptions`: this metric counts the total number of subscriptions to NRF; the metric stores the query type (`subscribe`, `update`, `unsubscribe`), the requester NF type and the result (`SUCCESS`, `FAILURE`)
* `nrf_nf_instances`: this metric counts the total number of queries to NRF for other NFs; the metric stores the requester NF type, the target NF type and the result (`SUCCESS`, `FAILURE`).

### CODE REVIEW

File: producer/nf_discovery.go [hunk 1]
```diff
@@ -35,14 +37,17 @@ func HandleNFDiscoveryRequest(request *httpwrapper.Request) *httpwrapper.Respons
...
 		return httpwrapper.NewResponse(http.StatusOK, nil, response)
 	} else if problemDetails != nil {
+		stats.IncrementNrfNfInstancesStats(fmt.Sprint(request.Query["requester-nf-type"][0]), fmt.Sprint(request.Query["target-nf-type"][0]), "SUCCESS")
 		return httpwrapper.NewResponse(int(problemDetails.Status), nil, problemDetails)
 	}
 	problemDetails = &models.ProblemDetails{
 		Status: http.StatusForbidden,
 		Cause:  "UNSPECIFIED",
 	}
+	stats.IncrementNrfNfInstancesStats(fmt.Sprint(request.Query["requester-nf-type"][0]), fmt.Sprint(request.Query["target-nf-type"][0]), "SUCCESS")
```
Comment (thakurajayL) [inline producer/nf_discovery.go:50]: if request.Query["requester-nf-type"][0] field is missing in request thenit may lead to crash.

File: producer/nf_discovery.go [hunk 2]
```diff
@@ -31,18 +33,22 @@ func HandleNFDiscoveryRequest(request *httpwrapper.Request) *httpwrapper.Respons
...
 	response, problemDetails := NFDiscoveryProcedure(request.Query)
+	requesterNfType, targetNfType := GetRequesterAndTargetNfTypeGivenQueryParameters(request.Query)
 	// Send Response
 	// step 4: process the return value from step 3
 	if response != nil {
 		// status code is based on SPEC, and option headers
+		stats.IncrementNrfNfInstancesStats(requesterNfType, targetNfType, "SUCCESS")
 		return httpwrapper.NewResponse(http.StatusOK, nil, response)
 	} else if problemDetails != nil {
+		stats.IncrementNrfNfInstancesStats(requesterNfType, targetNfType, "SUCCESS")
```
Comment (thakurajayL) [inline producer/nf_discovery.go:44]: Should this be failure?

Comment (thakurajayL) [reply producer/nf_discovery.go:44]: Please update the PR with changes.

File: producer/nf_discovery.go [hunk 3]
```diff
@@ -31,18 +33,22 @@ func HandleNFDiscoveryRequest(request *httpwrapper.Request) *httpwrapper.Respons
...
 		return httpwrapper.NewResponse(http.StatusOK, nil, response)
 	} else if problemDetails != nil {
+		stats.IncrementNrfNfInstancesStats(requesterNfType, targetNfType, "FAILURE")
 		return httpwrapper.NewResponse(int(problemDetails.Status), nil, problemDetails)
 	}
 	problemDetails = &models.ProblemDetails{
 		Status: http.StatusForbidden,
 		Cause:  "UNSPECIFIED",
 	}
+	stats.IncrementNrfNfInstancesStats(requesterNfType, targetNfType, "SUCCESS")
```
Comment (ajaythakurintel) [inline producer/nf_discovery.go:51]: This should be failure since http response code is forbidden.

Comment (ajaythakurintel) [inline producer/nf_discovery.go:51]:
```suggestion
	stats.IncrementNrfNfInstancesStats(requesterNfType, targetNfType, "FAILURE")
```

Comment (thakurajayL) [reply producer/nf_discovery.go]: Committed change.

**Review by thakurajayL — APPROVED** [review]

**Pull request merged**